### PR TITLE
types: migrate test-infra to tesify for const_test.go

### LINKED ISSUE #27996: migrate test-infra to testify for `const_test.go`

### PR DESCRIPTION
**Pull request by tisonkun**
Signed-off-by: tison <wander4096@gmail.com>

### What problem does this PR solve?

Issue Number: close #27996

### Release note

```release-note
None
```

### CODE REVIEW

File: types/const_test.go [hunk 1]
```diff
@@ -16,59 +16,17 @@ package types_test
...
+	t.Parallel()
 
-type testMySQLConstSuite struct {
-	cluster testutils.Cluster
-	store   kv.Storage
-	dom     *domain.Domain
-	*parser.Parser
-}
-
-var mockTikv = flag.Bool("mockTikv", true, "use mock tikv store in executor test")
```
Comment (tisonkun) [inline types/const_test.go:42]: We don't pass this flag any longer.

**Review by mmyj — APPROVED** [review]
lgtm

**Review by xhebox — APPROVED** [review]

**Pull request merged**